### ISSUE TITLE
Query: Regarding setting the new `which-key-replacement-alist' option

### ISSUE THREAD

**Issue by doolio** [justbur/emacs-which-key#165]
Firstly, thank you for this great package. I'm quite early, on my journey with Emacs and this package has gone someway in easing the path. Secondly, I'm an aspiring programmer so excuse my Emacs Lisp ignorance. This is a snippet from the `:config` part of a `which-key` `use-package` declaration in my current init.el. 

```Emacs Lisp
(setq 
which-key-key-replacement-alist
'(("<\\([[:alnum:]-]+\\)>" . "\\1")
  ("up"                    . "↑")
  ("right"                 . "→")
  ("down"                  . "↓")
  ("left"                  . "←"))
which-key-description-replacement-alist
'(("Prefix Command" . "prefix")
  ("\\`\\?\\?\\'"   . "λ")
  ("/body\\'"       . "|=")
  ("projectile-"    . "proj-")
  ("magit-"         . "ma-")))
```

With the latest `which-key` release these options are marked as obsolete and replaced by the `which-key-replacement-alist` option. I would therefore like to update this snippet to use the new alist option along the following lines taking example elements from the README and docstring. 

```Emacs Lisp
(setq 
which-key-replacement-alist
'((("<\\([[:alnum:]-]+\\)>" . nil) . ("\\1" . nil))
 ((nil . "Prefix Command") . (nil . "prefix"))))
```
which causes an (obvious?) error. I note the default value for this alist doesn't have the same number of inner cons cells as given above and in the README - why is that? My question therefore, is a form like this possible or do we have to implement this option using either the `push` function or `add-to-list` function as detailed in the README. Thank you for your time.

**Comment by justbur** [comment]
No, you should be able to do this. I don't see what's causing the error for you. I translated your original example like this and it seems to work fine. 

```elisp
(setq which-key-replacement-alist
      '((("<\\([[:alnum:]-]+\\)>") . ("\\1"))
        (("up") . ("↑"))
        (("right") . ("→"))
        (("down") . ("↓"))
        (("left") . ("←"))
        ((nil . "Prefix Command") . (nil . "prefix"))
        ((nil . "\\`\\?\\?\\'")   . (nil . "λ"))
        ((nil . "/body\\'")       . (nil . "|="))
        ((nil . "projectile-")    . (nil . "proj-"))
        ((nil . "magit-")         . (nil . "ma-"))))
```

**Comment by doolio** [comment]
You're quite right. My issue was a conflict between another package [validate](https://github.com/Malabarba/validate.el) and which-key. In my original snippet I had in fact `validate-setq` rather than `setq`. I omitted this fact as I believed my issue was due to me incorrectly configuring which-key and not my use of validate. It seems for reasons I don't yet understand configuring which-key to use validate (perhaps because it was inside the `:config` part of the use-package declaration) like this means your solution above does not work. If I use `setq` instead of `validate-setq` it works. Therefore, we can consider this 'issue' closed. Apologies for this wild goose chase, and thanks for your time.
